pkg/conditions/replicasready/update.go

```diff
@@ -0,0 +1,47 @@
+package replicasready
+
+import (
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+	capiconditions "sigs.k8s.io/cluster-api/util/conditions"
+)
+
+func update(machinePool *capiexp.MachinePool) {
+	// If value is not specified when MachinePool CR is created, the default
+	// value is ensured in azure-admission-controller.
+	desiredReplicas := *machinePool.Spec.Replicas
+
+	if desiredReplicas > machinePool.Status.Replicas {
```

Comment:
Is `>` the right operator instead of `!=` ?

Reply:
I was thinking about that, and the scenario I had in mind is when we scale up node pool for the upgrade (by doubling the nodes), so in that case we have:

- With `desiredReplicas != machinePool.Status.Replicas`, which is more strict check: for the above scenario the condition would be true, so `ReplicasReady` would be set to false. We are doubling the number of nodes, and at some point we get to have more than we defined in the CR spec - but due `!=` check `ReplicasReady` is false, even though we have more replicas than we actually need per spec.
- With `desiredReplicas > machinePool.Status.Replicas`, which is more loose check: in the above scenario, when we scale up, and we have more nodes than specified, `ReplicasReady` stays true.

Upstream is not doing any kind of desired vs. actual check here, even if let's say `3` replicas is set in CR Spec, if only 1 is deployed, and that single one is ready with nodeRef set, `ReplicasReady` is true. This is bad IMO, as is does not reflect correctly the state of the system (but then again, `MachinePool` is in experimental subpackage, so possibly it's not totally complete implementation).